Microsoft.Azure.Cosmos/src/Resource/CosmosException.cs

```diff
@@ -113,7 +113,7 @@ public class CosmosException : Exception
         /// <summary>
         /// Gets the internal headers
         /// </summary>
-        internal virtual ResponseHeaders Headers { get; }
+        internal virtual Headers Headers { get; }
```

Comment:
Having just Headers is confusing. 
```suggestion
        internal virtual Headers ResponseHeaders { get; }
```